documentation/deliberately_vulnerable_flask_app/test_pysa_output.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import os.path
+
+
+class PysaIssuesMismatchException(Exception):
+    """
+    Custom exception to raise when there is a mismatch with the issues
+    pysa generates and those in true_issues.json
+    """
+
+    pass
+
+
+def analyse_pysa_run() -> None:
+    """
+    Compares the issues pysa found in taint-output.json with those specified in
+    true_issues.json. Raises exceptions when files are missing or there are
+    mismatches.
+    """
+    # Check if we have both the generated and predefined files for comparison
+    if not os.path.isfile("./pysa-runs/taint-output.json"):
+        # output is not generated yet, error during compilation/generation
+        raise FileNotFoundError(
+            "./pysa-runs/taint-output.json file that should be generated by pysa is not found."
+        )
+    if not os.path.isfile("./true_issues.json"):
+        raise FileNotFoundError(
+            "./true_issues.json file that specifies issues to expect is not found"
+        )
+
+    # Parse the generated file data
+    taint_output_file: TextIOWrapper = open("./pysa-runs/taint-output.json", "r")
+    taint_output_contents: str = taint_output_file.read()
+    taint_output_file.close()
+
+    # Parse the predefined true_issues.json file data
+    true_issues: list = json.load(open("true_issues.json", "r"))["issues"]
+    true_issues_count: int = len(true_issues)
+    true_issues_detection_list: list = []
+
+    print("Expecting {} issues".format(true_issues_count))
+
+    # Go over each line in taint-output.json and check if the issues detected
+    # are the ones specified in true_issues.json
+    issue_counter: int = 0
+    false_positives: list = []
+    for line in map(lambda j: json.loads(j), taint_output_contents.split("\n")[1:]):
```

Comment:
Probably this parsing will need some changes if we read the output from  pyre with `subprocess.check_output(command, text=True)` as suggested above. My suggestion would be to refactor this a bit and create a separate function which takes the results of `subprocess.check_output(command, text=True)` and parses it to keep only some useful keys (e.g. callable, filename, I would also keep `code` which identify the issue kind) and then return a list of dict containing the parsed issues.
We can have a separate function which take care of comparing the expected issues with the actual ones and if they are different raise an error and save the actual results in a file called `results.actual`.
It's then pretty trivial for the user to compare the `results.actual` with the `results.expected` files and see whats the issue